trie.go

```diff
@@ -61,7 +61,10 @@ func (t *Trie) Add(key string, meta interface{}) *Node {
 // with `key`.
 func (t *Trie) Find(key string) (*Node, error) {
 	node := t.nodeAtPath(key)
-	node = node.Children()[nul]
+
+	if node != nil {
```

Comment:
Wasn't quite that easy because the !node.term check needs to be against the traversed node as well.